middleman-core/lib/middleman-core/util.rb

```diff
@@ -1,5 +1,4 @@
-# For instrumenting
-require 'active_support/notifications'
+require 'active_support/all'
```

Comment:
Is this change necessary/expected?

Reply:
Yes, we need it. Otherwise, we will have an error when with activesupport 7.0.3.1. On 5.x (master branch), this was already done in #2256 3 years ago.

### Error log with activesupport 7.0.3.1
https://github.com/middleman/middleman/runs/7314924787?check_suite_focus=true

```
Failures:
  1) Middleman::Util::asset_url returns a resource url if given a resource's destination path
     Failure/Error: super
     NameError:
       uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
     # ./lib/middleman-core/logger.rb:35:in `block in add'
     # ./lib/middleman-core/logger.rb:34:in `synchronize'
     # ./lib/middleman-core/logger.rb:34:in `add'
     # ./lib/middleman-core/extension_manager.rb:57:in `activate'
     # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
     # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
     # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
     # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
     # ./lib/middleman-core/callback_manager.rb:57:in `execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
     # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
     # ./lib/middleman-core/application.rb:272:in `initialize'
     # ./spec/middleman-core/util_spec.rb:152:in `new'
     # ./spec/middleman-core/util_spec.rb:152:in `block (3 levels) in <top (required)>'
  2) Middleman::Util::asset_url returns a resource url if given a resources path
     Failure/Error: super
     NameError:
       uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
     # ./lib/middleman-core/logger.rb:35:in `block in add'
     # ./lib/middleman-core/logger.rb:34:in `synchronize'
     # ./lib/middleman-core/logger.rb:34:in `add'
     # ./lib/middleman-core/extension_manager.rb:57:in `activate'
     # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
     # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
     # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
     # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
     # ./lib/middleman-core/callback_manager.rb:57:in `execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
     # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
     # ./lib/middleman-core/application.rb:272:in `initialize'
     # ./spec/middleman-core/util_spec.rb:160:in `new'
     # ./spec/middleman-core/util_spec.rb:160:in `block (3 levels) in <top (required)>'
  3) Middleman::Util::asset_url returns path relative to the provided current_resource
     Failure/Error: super
     NameError:
       uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
     # ./lib/middleman-core/logger.rb:35:in `block in add'
     # ./lib/middleman-core/logger.rb:34:in `synchronize'
     # ./lib/middleman-core/logger.rb:34:in `add'
     # ./lib/middleman-core/extension_manager.rb:57:in `activate'
     # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
     # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
     # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
     # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
     # ./lib/middleman-core/callback_manager.rb:57:in `execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
     # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
     # ./lib/middleman-core/application.rb:272:in `initialize'
     # ./spec/middleman-core/util_spec.rb:105:in `new'
     # ./spec/middleman-core/util_spec.rb:105:in `block (3 levels) in <top (required)>'
  4) Middleman::Util::asset_url returns a resource url when asset_hash is on
     Failure/Error: super
     NameError:
       uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
     # ./lib/middleman-core/logger.rb:35:in `block in add'
     # ./lib/middleman-core/logger.rb:34:in `synchronize'
     # ./lib/middleman-core/logger.rb:34:in `add'
     # ./lib/middleman-core/extension_manager.rb:57:in `activate'
     # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
     # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
     # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
     # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
     # ./lib/middleman-core/callback_manager.rb:57:in `execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
     # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
     # ./lib/middleman-core/application.rb:272:in `initialize'
     # ./spec/middleman-core/util_spec.rb:167:in `new'
     # ./spec/middleman-core/util_spec.rb:167:in `block (3 levels) in <top (required)>'
  5) Middleman::Util::asset_url when relative is true raises error if not given a current_resource
     Failure/Error: super
     NameError:
       uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
     # ./lib/middleman-core/logger.rb:35:in `block in add'
     # ./lib/middleman-core/logger.rb:34:in `synchronize'
     # ./lib/middleman-core/logger.rb:34:in `add'
     # ./lib/middleman-core/extension_manager.rb:57:in `activate'
     # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
     # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
     # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
     # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
     # ./lib/middleman-core/callback_manager.rb:57:in `execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
     # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
     # ./lib/middleman-core/application.rb:272:in `initialize'
     # ./spec/middleman-core/util_spec.rb:114:in `new'
     # ./spec/middleman-core/util_spec.rb:114:in `block (4 levels) in <top (required)>'
  6) Middleman::Util::asset_url when relative is true returns path relative to the provided current_resource
     Failure/Error: super
     NameError:
       uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
     # ./lib/middleman-core/logger.rb:35:in `block in add'
     # ./lib/middleman-core/logger.rb:34:in `synchronize'
     # ./lib/middleman-core/logger.rb:34:in `add'
     # ./lib/middleman-core/extension_manager.rb:57:in `activate'
     # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
     # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
     # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
     # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
     # ./lib/middleman-core/callback_manager.rb:57:in `execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
     # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
     # ./lib/middleman-core/application.rb:272:in `initialize'
     # ./spec/middleman-core/util_spec.rb:114:in `new'
     # ./spec/middleman-core/util_spec.rb:114:in `block (4 levels) in <top (required)>'
  7) Middleman::Util::asset_url when relative is true when the asset is stored in the same directory as current_resource returns path relative to the provided current_resource
     Failure/Error: super
     NameError:
       uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
     # ./lib/middleman-core/logger.rb:35:in `block in add'
     # ./lib/middleman-core/logger.rb:34:in `synchronize'
     # ./lib/middleman-core/logger.rb:34:in `add'
     # ./lib/middleman-core/extension_manager.rb:57:in `activate'
     # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
     # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
     # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
     # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
     # ./lib/middleman-core/callback_manager.rb:57:in `execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
     # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
     # ./lib/middleman-core/application.rb:272:in `initialize'
     # ./spec/middleman-core/util_spec.rb:114:in `new'
     # ./spec/middleman-core/util_spec.rb:114:in `block (4 levels) in <top (required)>'
  8) Middleman::Util::asset_url when http_prefix is activated returns path with http_prefix pre-pended if resource is not found
     Failure/Error: super
     NameError:
       uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
     # ./lib/middleman-core/logger.rb:35:in `block in add'
     # ./lib/middleman-core/logger.rb:34:in `synchronize'
     # ./lib/middleman-core/logger.rb:34:in `add'
     # ./lib/middleman-core/extension_manager.rb:57:in `activate'
     # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
     # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
     # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
     # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
     # ./lib/middleman-core/callback_manager.rb:57:in `execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
     # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
     # ./lib/middleman-core/application.rb:272:in `initialize'
     # ./spec/middleman-core/util_spec.rb:87:in `new'
     # ./spec/middleman-core/util_spec.rb:87:in `block (4 levels) in <top (required)>'
  9) Middleman::Util::asset_url when http_prefix is activated returns path with http_prefix pre-pended if resource is found
     Failure/Error: super
     NameError:
       uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
     # ./lib/middleman-core/logger.rb:35:in `block in add'
     # ./lib/middleman-core/logger.rb:34:in `synchronize'
     # ./lib/middleman-core/logger.rb:34:in `add'
     # ./lib/middleman-core/extension_manager.rb:57:in `activate'
     # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
     # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
     # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
     # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
     # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
     # ./lib/middleman-core/callback_manager.rb:57:in `execute'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
     # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
     # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
     # ./lib/middleman-core/application.rb:272:in `initialize'
     # ./spec/middleman-core/util_spec.rb:87:in `new'
     # ./spec/middleman-core/util_spec.rb:87:in `block (4 levels) in <top (required)>'
  10) Middleman::Util::find_related_files Finds partials possibly related to ERb files
      Failure/Error: super
      NameError:
        uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
      # ./lib/middleman-core/logger.rb:35:in `block in add'
      # ./lib/middleman-core/logger.rb:34:in `synchronize'
      # ./lib/middleman-core/logger.rb:34:in `add'
      # ./lib/middleman-core/extension_manager.rb:57:in `activate'
      # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
      # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
      # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
      # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
      # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
      # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
      # ./lib/middleman-core/callback_manager.rb:57:in `execute'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
      # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
      # ./lib/middleman-core/application.rb:272:in `initialize'
      # ./spec/middleman-core/util_spec.rb:181:in `new'
      # ./spec/middleman-core/util_spec.rb:181:in `block (3 levels) in <top (required)>'
  11) Middleman::Util::find_related_files Finds partials possible related to Scss files
      Failure/Error: super
      NameError:
        uninitialized constant ActiveSupport::LoggerThreadSafeLevel::IsolatedExecutionState
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:21:in `local_level'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:37:in `level'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/activesupport-7.0.3.1/lib/active_support/logger_thread_safe_level.rb:54:in `add'
      # ./lib/middleman-core/logger.rb:35:in `block in add'
      # ./lib/middleman-core/logger.rb:34:in `synchronize'
      # ./lib/middleman-core/logger.rb:34:in `add'
      # ./lib/middleman-core/extension_manager.rb:57:in `activate'
      # ./lib/middleman-core/extensions.rb:120:in `block in auto_activate'
      # ./lib/middleman-core/extensions.rb:117:in `auto_activate'
      # ./lib/middleman-core/extension_manager.rb:28:in `auto_activate'
      # ./lib/middleman-core/extension_manager.rb:20:in `block (2 levels) in initialize'
      # ./lib/middleman-core/callback_manager.rb:57:in `instance_exec'
      # ./lib/middleman-core/callback_manager.rb:57:in `block in execute'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `each'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:1316:in `traverse_depth_first'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/hamster-3.0.0/lib/hamster/vector.rb:431:in `each'
      # ./lib/middleman-core/callback_manager.rb:57:in `execute'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_reference.rb:43:in `send_to'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/call_with.rb:76:in `call_with'
      # /home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/contracts-0.13.0/lib/contracts/method_handler.rb:138:in `block in redefine_method'
      # ./lib/middleman-core/callback_manager.rb:28:in `block in install_methods!'
      # ./lib/middleman-core/application.rb:272:in `initialize'
      # ./spec/middleman-core/util_spec.rb:181:in `new'
      # ./spec/middleman-core/util_spec.rb:181:in `block (3 levels) in <top (required)>'
Finished in 0.53446 seconds (files took 0.82029 seconds to load)
67 examples, 11 failures
Failed examples:
rspec ./spec/middleman-core/util_spec.rb:149 # Middleman::Util::asset_url returns a resource url if given a resource's destination path
rspec ./spec/middleman-core/util_spec.rb:157 # Middleman::Util::asset_url returns a resource url if given a resources path
rspec ./spec/middleman-core/util_spec.rb:101 # Middleman::Util::asset_url returns path relative to the provided current_resource
rspec ./spec/middleman-core/util_spec.rb:165 # Middleman::Util::asset_url returns a resource url when asset_hash is on
rspec ./spec/middleman-core/util_spec.rb:137 # Middleman::Util::asset_url when relative is true raises error if not given a current_resource
rspec ./spec/middleman-core/util_spec.rb:117 # Middleman::Util::asset_url when relative is true returns path relative to the provided current_resource
rspec ./spec/middleman-core/util_spec.rb:130 # Middleman::Util::asset_url when relative is true when the asset is stored in the same directory as current_resource returns path relative to the provided current_resource
rspec ./spec/middleman-core/util_spec.rb:96 # Middleman::Util::asset_url when http_prefix is activated returns path with http_prefix pre-pended if resource is not found
rspec ./spec/middleman-core/util_spec.rb:92 # Middleman::Util::asset_url when http_prefix is activated returns path with http_prefix pre-pended if resource is found
rspec ./spec/middleman-core/util_spec.rb:188 # Middleman::Util::find_related_files Finds partials possibly related to ERb files
rspec ./spec/middleman-core/util_spec.rb:196 # Middleman::Util::find_related_files Finds partials possible related to Scss files
Randomized with seed 34748
[...]
/home/runner/work/middleman/middleman/vendor/bundle/ruby/2.7.0/gems/rspec-core-3.10.1/exe/rspec --pattern spec/\*\*/\*_spec.rb --color --format documentation failed
```